Automattic/wp-calypso
# Remove legacy checklist section

#### Changes proposed in this Pull Request

Removes the no longer used checklist section after enabling My Home for everyone. 

After auditing and updating all the places that were still linking to `/checklist` in #40952, #40956 and D41640-code, the checklist section should be now safe to remove.

#### Testing instructions

- Make sure the `/checklist` path does not load anything.
- Make sure there are no regressions (no dead links to `/checklist`) when creating a new site (you should land in `/home`) or when purchasing a Jetpack plan in Calypso (you should land in `/plans/my-plan`).
- Make sure the site setup checklist is displayed correctly in My Home for new created sites.
- Make sure the Jetpack checklist is displayed correctly in My Plan for new purchased plans.

Fixes #40646
